rust/lance-io/src/object_store.rs

```diff
@@ -384,6 +385,17 @@ impl ObjectStore {
     /// cached metadata. By passing in the known size, we can skip a HEAD / metadata
     /// call.
     pub async fn open_with_size(&self, path: &Path, known_size: usize) -> Result<Box<dyn Reader>> {
+        // If we know the file is really small, we can read the whole thing
+        // as a single request.
+        if known_size <= self.block_size {
```

Comment:
What's the reason for basing it on the block size? I would guess the break even point for one request vs several is probably somewhere higher than 64K, but it would require some experimentation to know.

Reply:
We could probably do 2x block size for these kinds of requests. Can experiment later.